### PR TITLE
Page flow

### PR DESCRIPTION
**Pull request by stecychirinda**
### Description
This pull request implements the remaining pages required for Milestone 1, including the Home page, Buddy Pairing page, and Verification Code page. The changes made in this PR are as follows:
For the Home page:

- Lifted the state of profile data to use it in the Homepage.
- Set up the HomePage.
- Linked the HomePage with the Buddy finding page.
- I added the meeting point option to ensure users meet in open safe environments.

For the Buddy Pairing page:

- Set up the Buddy Finding Page.
- Hardcoded two buddy match info in list format.
- Linked the page to Verification code page.

For the Verification page:

- Set up the Verification page.
- Implemented input to be block (each code digit in its own input block).

Note that styling using Tailwind will be implemented in a later PR.

### Milestones
This work contributes to Milestone 1, which includes the following features:

- Homepage + Buddy Finder
- Destination search bar (e.g., Library, Dining Hall)
- "Find Walking Buddies" button dynamically matches users (display page until implementation of logic)

### Resources
The design of these pages draws inspiration from the Uber homepage and verification code, as well as other similar apps.

### Test Plan
Attached are some images of the general pages without styling. These changes were tested by verifying that the pages load correctly and link to each other as expected. Edge cases such as invalid user input will be tested in future PRs.

Screenshot 2025-06-27 at 9 48 46 AM
Screenshot 2025-06-27 at 9 49 04 AM
Screenshot 2025-06-27 at 9 49 16 AM

### CODE REVIEW

**Comment by jzysak25** [comment]
When it comes to setting a responsive dimension, I believe more in % instead of using vh and vw directly.
The reason is % refers to the width and height of parental container, while vh and vw directly give you a way to hardcode the dimension of a component from the viewport, ignoring the whole DOM tree. This usually leads to overflow of the content, when you start to deal more component in the page, you will notice the page might not layout exactly as you would like to because you have some hidden component directly occupying some part of the page with vh and vw.

**Review by jzysak25 — CHANGES_REQUESTED** [review]

File: Backend/index.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const PORT = 3000;
 
 app.use(
   cors({
-    origin: "http://localhost:5179",
```
[the remark this answers is not in the document]
Comment (stecychirinda) [reply Backend/index.js:11]: Noted, will do so.
